IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -744,5 +745,17 @@ - (MSIDCacheKey *)overrideTokenKey:(MSIDCacheKey *)key
     return key;
 }
 
+- (NSString *)extractAppKey:(NSString *)cacheKeyString
+{
+    if ([NSString msidIsStringNilOrBlank:cacheKeyString]) return nil;
+    
+    NSArray *components = [cacheKeyString componentsSeparatedByString:@"|"];
```

Comment:
will this also work with ADAL cache items? I think there were multiple pipe (|) separated components in service, so the count won't be 2 in that case.

Reply:
good catch, fixed it and also added unit test cases :)

Reply:
👍